pyspades/player.py

```diff
@@ -1217,11 +1217,21 @@ def continue_map_transfer(self) -> None:
     def send_data(self, data):
         self.protocol.transport.write(data, self.address)
 
-    def send_chat(self, value: str, global_message: bool = False) -> None:
+    def send_chat(self, value: str, global_message: bool = False, custom_type: int = CHAT_ALL) -> None:
         if self.deaf:
             return
         chat_message = loaders.ChatMessage()
-        if not global_message:
+        if custom_type > 2 and "client" in self.client_info:
+            if self.client_info["client"] == "BetterSpades":
+                chat_message.chat_type = custom_type
+
+            elif self.client_info["client"] == "OpenSpades":
```

Comment:
iirc the capability to send custom message types in OS predates the ability to query for client version.
Because of this, I would just unconditionally send the markup in front. It doesn't hurt to just send it either way.

Reply:
You mean without checking OS version, or for all clients?

Reply:
I would send it for all clients that do not support the chat type extension?